### DISCUSSION TITLE
New year release of 0.2.0

### DISCUSSION THREAD

**Discussion by kcubeterm** [kcubeterm/achoz#16] [Announcements]
No matter what we will release achoz on the eve of new year for sure. This release will have following features and usable and functional. 

- [ ] Logger (Logs helps in figure out issue)
- [ ] Test unit
- [ ] PDF viewer
- [x] Video preview 
- [x] Audio preview
- [x] Image preview
- [ ] Add support of Meilisearch as well 
- [x] Efficient indexing and more robust crawler. 

Why need meilisearch ? 
Meilisearch support arm and aarch64 devices unlike typesense. With meilisearch achoz will run on Android device with Termux.

**Comment by cauerego** [comment]
i would choose "efficient indexing and more robust crawler" to come next! 😁

**Reply by cauerego** [reply]
except, a "general viewer" is very important!

let me explain...

in theory, we will always have terminal (at least SSH) access to the server, right? so perhaps every search result should link to some direct way of opening files... regardless of viewers (such as pdf viewer).

**Reply by kcubeterm** [reply]
yes, every result is linked with file but indirectly. client have to request file with its id from server. this function is responsible to serve file. i used id to request files so that any type of command injection or unindexed file request could be avoided.
 
https://github.com/kcubeterm/achoz/blob/86adaa0d8855221df77ebdc8eb2b7a5832398c10/server.js#L68

**Reply by kcubeterm** [reply]
also, do you mean that search result files could be open with another application in device as well ? then that could be done with xdg but only if achoz is hosted in same device

**Reply by cauerego** [reply]
if we have SSH access, it's more a matter of linking to a protocol which can be opened either locally or remotely, it shouldn't matter.

the injection prevention is fine. and showing the path to the file is already a nice touch... at least we can already manually open it.

in other words, if we have SSH access, then we should have some way to directly open the file from the web browser...

but on a second thought, to make everything easier: 

just offer a link to open the file!

let the browser decide what to do with it.

**Comment by cauerego** [comment]
you sure meilisearch is needed for running on android?! 🤔

**Reply by kcubeterm** [reply]
we just need a search engine which support arm and aarch64 and as of now typesense dont support

**Reply by cauerego** [reply]
i can't imagine how any code written in the past decade wouldn't be able to compile in any end user processor... what do you mean, it doesn't support arm?

**Reply by cauerego** [reply]
got my answer directly from the source... Kishore:

>  Typesense and its dependencies make some assumptions about an X86 architecture which must be addressed before we can build on ARM.

but even before that, he gave us another idea:

> run Typesense as a service and use the Java client for access.

makes sense? 😁

**Reply by kcubeterm** [reply]
Make sense offcourse and it has already implemented. If this would be case then user just need to tweak "TypesenseHost" and "TypesenseApi" in config.json

**Reply by cauerego** [reply]
to me, i still can't make much sense why they needed to make those assumptions... 🤣

feels way too much like over optimization.
